src/Kernel32.Shared/Kernel32.Helpers.cs

```diff
@@ -47,5 +48,60 @@ public static Win32ErrorCode GetLastError()
         {
             return (Win32ErrorCode)Marshal.GetLastWin32Error();
         }
+
+        public static unsafe uint WriteFile(SafeObjectHandle hFile, void* lpBuffer, uint nNumberOfBytesToWrite)
```

Comment:
Yes, advanced scenarios like reading directly into an array of structures or writing a pre-defined structure would be blocked.

Another common scenario is writing an array but not from the first element. Like if another call to `WriteFile` returned a number of bytes written too small you want to finish the write, passing the same buffer, moving just the pointer inside. It's actually a potential (Easy to forget) occurence  with pipes :

> When writing to a non-blocking, byte-mode pipe handle with insufficient buffer space, WriteFile returns TRUE with *lpNumberOfBytesWritten < nNumberOfBytesToWrite.

A `byte[]` would force all of theses to make an allocation + a memory copy. Other helpers can take care of the array version (Via an `ArraySegment` here).